lalrpop/src/tok/mod.rs

```diff
@@ -564,6 +580,49 @@ impl<'input> Tokenizer<'input> {
         }
     }
 
+    fn block_comment(&mut self, idx0: usize) -> Result<(), Error> {
+        #[derive(PartialEq, Eq)]
+        enum State {
+            Initial,
+            Slash,
+            Star,
+            Complete,
+        }
+
+        let mut depth = 1;
+        let mut state = State::Initial;
+
+        let end_of_comment = |c: char| {
+            state = match state {
+                State::Initial if c == '/' => State::Slash,
+                State::Initial if c == '*' => State::Star,
+                State::Slash if c == '*' => {
+                    depth += 1;
+                    State::Initial
+                }
+                State::Star if c == '/' => {
+                    depth -= 1;
+                    if depth == 0 {
+                        State::Complete
+                    } else {
+                        State::Initial
+                    }
+                }
+                _ => State::Initial,
```

Comment:
If you are in the star state, and see another star, shouldn't you stay in the star state?

Consider these three examples which are valid rust comments. Maybe elevate at least into a test.

```rust    
/***/
/*****/
/* **/
```

Reply:
Not that we need to be fully compatible, but here is the grammar for comments: https://doc.rust-lang.org/reference/comments.html

Reply:
Along the same lines, shouldn't you stay in the Slash state if you're in the Slash state and see another slash?

I believe that `/*//**/*/` is a valid comment that I don't think this would parse correctly.

I wonder if the whole thing would be a little cleaner and more idiomatic as something like:

```
match (state, c) {
    (State::Initial, '/') => State::Slash
    // etc
```

Reply:
I *think* with the above suggestion, you can write the final clause as `(_,c) if c != '/' && c != '*''`, and get a compiler guarantee that all the '/' and '*' cases are covered

Reply:
I was not able to get this to compile without a warning, so I reverted to just using the `_` match.